python/ray/tune/tests/test_result_grid.py

```diff
@@ -14,11 +14,15 @@ def f(config):
             with open(path, "w") as f:
                 f.write(json.dumps({"step": 0}))
 
-    analysis = tune.run(f)
+    analysis = tune.run(f, config={"a": 1})
```

Comment:
Not needed for this PR, but should we move `test_result_grid` to ray.ml.tests and have it use `Tuner` instead of `tune.run`.